Mage.Sets/src/mage/cards/l/Lifeline.java

```diff
@@ -0,0 +1,146 @@
+import mage.abilities.common.DiesCreatureTriggeredAbility;
+import mage.abilities.common.delayed.AtTheBeginOfNextEndStepDelayedTriggeredAbility;
+import mage.abilities.condition.common.PermanentsOnTheBattlefieldCondition;
+import mage.abilities.decorator.ConditionalTriggeredAbility;
+import mage.abilities.effects.Effect;
+import mage.abilities.effects.OneShotEffect;
+import mage.cards.Card;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.cards.Cards;
+import mage.cards.CardsImpl;
+import mage.constants.CardType;
+import mage.constants.Outcome;
+import mage.constants.Zone;
+import mage.filter.common.FilterCreaturePermanent;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+
+/**
+ *
+ * @author HCrescent
+ */
+public class Lifeline extends CardImpl {
+    
+private static final FilterCreaturePermanent filter = new FilterCreaturePermanent("creature");
+
+    public Lifeline(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ARTIFACT}, "{5}");
+                
+
+        // Whenever a creature dies, if another creature is on the battlefield, return the first card to the battlefield under its owner's control at the beginning of the next end step.
+        Ability ability = new ConditionalTriggeredAbility(
+                            new DiesCreatureTriggeredAbility(
+                                new LifelineEffect(),
+                                false,
+                                false,
+                                true),
+                            new PermanentsOnTheBattlefieldCondition(filter, PermanentsOnTheBattlefieldCondition.CountType.MORE_THAN, 0, false),
+                            "Whenever a creature dies, if another creature is on the battlefield, return the first card to the battlefield under its owner's control at the beginning of the next end step.");
+        this.addAbility(ability);
+    }
+
+    public Lifeline(final Lifeline card) {
+        super(card);
+    }
+
+    @Override
+    public Lifeline copy() {
+        return new Lifeline(this);
+    }
+}
+class LifelineEffect extends OneShotEffect {
+    
+    public LifelineEffect() {
+        super(Outcome.Benefit);
+        this.staticText = "";
+    }
+    
+    public LifelineEffect(final LifelineEffect effect) {
+        super(effect);
+    }
+    
+    @Override
+    public LifelineEffect copy() {
+        return new LifelineEffect(this);
+    }
+    
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Cards cards = new CardsImpl();
+        Permanent permanent = game.getPermanentOrLKIBattlefield(this.getTargetPointer().getFirst(game, source));
+        if (permanent != null) {
+            //Tracking zone changes after first effect
+            permanent.setZoneChangeCounter(0, game);
+            cards.add(permanent);
+        }
+        Effect effect = new LifelineDelayedEffect();
+        effect.setValue("LifelineCards", cards);

```

Comment:
Maybe I'm wrong about the best practices but if I were implementing this I would be passing this to the LifelineDelayedEffect constructor rather then using the value system. I guess that's something of a stylistic choice though.